### PR TITLE
Multiple sites per page [PD-1174] and some accidental caching [PD-1146]

### PR DESCRIPTION
**Pull request by almathew**

### CODE REVIEW

**Comment by aspidites** [comment]
"and some accidental caching"?

**Comment by almathew** [comment]
I meant for it to be a separate branch.

File: myblocks/context_tools.py [hunk 1]
```diff
@@ -234,6 +234,12 @@ def get_query_string(request):
 
 
 @Memoized
```
Comment (aspidites) [inline myblocks/context_tools.py]: Can we use the built in [memoize decorator](https://docs.djangoproject.com/en/1.6/_modules/django/utils/functional/)? (that I mentioned about a week ago)

Comment (almathew) [inline myblocks/context_tools.py]: I tried messing around with it when I first started memoizing after you brought it up. There's an issue with the way the django's version handled hashing that was either causing requests to always match or be unique (I honestly forget which, but I think it was be unique).

File: myblocks/models.py [hunk 2]
```diff
@@ -502,9 +528,33 @@ def pixel_template(self):
...
+            return rendered_template
         context = self.context(request, **kwargs)
         template = Template(self.get_template(request))
-        return template.render(RequestContext(request, context))
+        rendered_template = template.render(RequestContext(request, context))
+        blocks_cache.set(key, rendered_template, settings.MINUTES_TO_CACHE*60)
+        return rendered_template
+
+    def render_cache_prefix(self, request):
+        page = '::'.join([str(self.pk), str(self.updated)])
```
Comment (aspidites) [inline myblocks/models.py]: modulo string formatting is faster (in python 2), and arguably clearer -> `'%s::%s' % (self.pk, self.updated)`. Also doesn't require multiple calls to `str`.

File: myblocks/context_tools.py (hunk 1 above)
Comment (aspidites) [inline myblocks/context_tools.py]: OK. Decorator class may be overkill. That you are overriding `__call__` seems like a good indicator that what you really want is a decorator function.

Comment (almathew) [inline myblocks/context_tools.py]: Copy + paste from here: https://wiki.python.org/moin/PythonDecoratorLibrary#Memoize

Comment (aspidites) [inline myblocks/context_tools.py]: [From that same page](https://wiki.python.org/moin/PythonDecoratorLibrary#Alternate_memoize_as_nested_functions):
```python
def memoize(obj):
     cache = obj.cache = {}
 
     @functools.wraps(obj)
     def memoizer(*args, **kwargs):
         key = str(args) + str(kwargs)
         if key not in cache:
             cache[key] = obj(*args, **kwargs)
         return cache[key]
     return memoizer
```

**Pull request merged**